src/crate/client/pep440.py

```diff
@@ -0,0 +1,582 @@
+# Vendored from `packaging` on 2021-30-04, via pandas.
+# - https://github.com/pypa/packaging/blob/ae891fd74/packaging/_structures.py
+# - https://github.com/pypa/packaging/blob/ae891fd74/packaging/version.py
+# - https://github.com/pandas-dev/pandas/blob/cf2b88b5ea/pandas/util/version/__init__.py
+
+# This file is dual licensed under the terms of the Apache License, Version
+# 2.0, and the BSD License. See the LICENSE file in the root of this repository
+# for complete details.
```

Comment:
If we use this license related header, we'd have to include the original license note to our LICENSE file afaik similar how pandas did it https://github.com/pandas-dev/pandas/commit/f40e58cc50abcd03efae372df0586467a957f695#diff-17d4b6290d4495e094bd6dc27902057dcb25ca7f73d595162899c22d1b95f43eR180. 

Or we keep the license header of the donald like numpy did https://github.com/numpy/numpy/pull/21000/files#diff-41a2b10dd4230acb81d1f9f7dedd8e8e6f6ef4f44d15f20105e9e82cf5a489ebR7.

Reply:
### Response
Thank you for your attention.

### Thoughts
I think it would be even better if someone would release and maintain a dedicated `versions`, or `pep440` package, that this code would not be vendored over and over again.

### Discovery
Better late than never: It turns out, that happened already!

- https://pypi.org/project/versions/
- https://pypi.org/project/pep440/

### Thoughts
Maybe better spend another iteration to re-use one of those packages instead?

Reply:
> Maybe better spend another iteration to re-use one of those packages instead?

Hm. While `versions` feels a bit too heavy, I think `pep440` is a bit too thin.

Reply:
Yes sounds good, lets use one of them.

Reply:
> > Maybe better spend another iteration to re-use one of those packages instead?
> 
> Hm. While `versions` feels a bit too heavy, I think `pep440` is a bit too thin.

Another one: https://pypi.org/project/pep440-version-utils/
But also not sure if this is really worth of a dependency. Maybe we should just fix the license header instead?

Reply:
> Another one: https://pypi.org/project/pep440-version-utils/.

Thanks, but this one actually _pulls in_ `packaging`, see [pyproject.toml#L16](https://github.com/m-vdb/pep440-version-utils/blob/10e1eeb8839c97b91442fe08c117ea02d278ba92/pyproject.toml#L16).

> But also not sure if this is really worth of a dependency.

Yeah, I think that was the same reasoning of the other maintainers.

> Maybe we should just fix the license header instead?

Sure, let's do it like that for now.

Reply:
I've reworked the patch to use the `_pep440.py` file from current NumPy [^1], as you suggested.

[^1]: https://github.com/numpy/numpy/blob/v1.25.0.dev0/numpy/compat/_pep440.py